src/lib.c

```diff
@@ -373,6 +371,49 @@ search_esp_filename(const char *filename, char *esp_filename)
 	esp_filename[0] = '\0';
 }
 
+static
+void save_esp_filename(const char *esp_filename) {
+    FILE *var2file = NULL;
+    FILE *output_file = NULL;
+    unsigned char buffer[1024];
+    size_t bytes_read, bytes_written;
+	bool fail = false;
+
+    var2file = fopen(VAR2FILE, "rb");
+    if (var2file == NULL) {
```

Comment:
I am not sure about the efivar coding style,m but we usually do if (!var2file)

Reply:
Done